test/check-trace.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+max_retries=50
+retries=0
+while true
+do
+    traces=$(curl -s "http://localhost:16686/api/traces?service=e2e-test")
+    numTraces=$(echo ${traces} | jq -r '.data' | jq length)
+    numSpans=$(echo ${traces}  | jq -r '.data[0].spans' | jq length)
+
+    # we expect at least one trace and at least two spans in the first trace
+    if (( numTraces > 0 )); then
+        echo "found at least one trace for the e2e-test service" >> ./test/test.log
+        if (( numSpans > 1 )); then
+            echo "found at least two spans in the trace" >> ./test/test.log
+            echo "✅ Traces found in the backend."
+            break
+        else
+            echo "not enough spans in the trace" >> ./test/test.log
```

Comment:
nit, and a completely optional one.  It *might* be useful to print the actual number of spans received.  Information like that is often useful in debugging test failures.

Reply:
Done!